bibliopixel/drivers/SPI/APA102.py

```diff
@@ -0,0 +1,27 @@
+from ... import gamma
+from .. channel_order import ChannelOrder
+from . base import SPIBase
+
+
+class APA102(SPIBase):
+    """Main driver for APA102 based LED strips on devices like the Raspberry Pi and BeagleBone"""
+
+    def __init__(self, num, gamma=gamma.APA102, **kwargs):
+        super().__init__(num, gamma=gamma, **kwargs)
+
+        # APA102 requires latch bytes at the end
+        self._latchBytes = (int(num // 64.0) + 1)
+
+    def _compute_packet(self):
+        self._render()
+        newBuf = [0xFF] * (self.bufByteCount() + self.numLEDs)
+        newBuf[1::4] = self._buf[0::3]
+        newBuf[2::4] = self._buf[1::3]
+        newBuf[3::4] = self._buf[2::3]
+        newBuf = [0, 0, 0, 0] + newBuf
+        newBuf.extend([0xFF, 0, 0, 0] * self._latchBytes)
```

Comment:
0. This fails in what at work we label as "NEFR" (Not Eligable For Review). As in, this is basically old code (which in this case I'm pretty sure *you* put there when you converted all the drivers) and not technically part of this commit. Though I admit, that confused the shit out of me when I was making these updates. Though, looking back at it... yeah, it needs to call render because that's what actually does the brightness and gamma calculations in driver_base. So everything should call render() and *then* do stuff to it after that.

1. Fix incoming.

2. What would you prefer. Makes sense to me.

3. TODO added. I had the same thought but at the time was trying to only clean up the SPI system and fix things that were broken, not change any existing code. Also, I'm going to rewrite the APA102 driver after 3.0 release (probably for a 3.0.x release) to add SK9822 (an APA clone) and better brightness control.